website/static/js/citationList.js

```diff
@@ -19,10 +20,169 @@ var formatCitation = function(style, data, format) {
+        }));
+
+    self.editing = ko.observable(false);
+
+    self.addAlternative = function() {
+        self.citations.push(new AlternativeModel());
+        self.editing(true);
+    };
+    self.apa = ko.observable();
+    self.mla = ko.observable();
+    self.chicago = ko.observable();
+};
+
+var AlternativeModel = function(citation, view) {
+    var self = this;
+
+    self.init = function() {
+        if (citation !== undefined) {
+            $.extend(self, citation);
+            self.name = ko.observable(citation.name);
+            self.text = ko.observable(citation.text);
+            self.id = citation.id;
+            self.view = ko.observable(view);
+            self.originalValues = {
+                name: citation.name,
+                text: citation.text
+            };
+        }
+        else {
+            self.name = ko.observable();
+            self.text = ko.observable();
+            self.view = ko.observable();
+            self.edit();
+        }
+
+        self.messages = ko.observableArray([]);
+    };
+
+    self.removeSelf = function(parent) {
+        if (self.originalValues !== undefined) {
+            bootbox.confirm({
+                title: 'Delete citation?',
+                message: ('Are you sure you want to remove this citation (<strong>' + $osf.htmlEscape(self.name()) + '</strong>)?'),
+                callback: function () {
+                    $osf.ajaxJSON('DELETE', $osf.apiV2Url('nodes/' + ctx.node.id + '/citations/' + self.id + '/'), {isCors: true}).done(function(response) {
+                        var index = parent.citations.indexOf(self);
+                        parent.citations.splice(index, 1);
+                        if (parent.editing()) {
+                            parent.editing(false);
+                        }
+                    }).fail(function(response){
+                        $osf.growl('Error:',
+                            'An unexpected error has occurred.  Please try again later.  If problem persists contact <a href="mailto: support@cos.io">support@cos.io</a>', 'danger'
+                        );
+                    });
+                },
+                buttons:{
+                    confirm:{
+                        label:'Delete',
+                        className:'btn-danger'
+                    }
+                }
+            });
+        }
+    };
+
+    self.cancel = function(parent) {
+        if (self.originalValues === undefined) {
+            var index = parent.citations.indexOf(self);
+            parent.citations.splice(index, 1);
+        }
+        else {
+            self.name(self.originalValues.name);
+            self.text(self.originalValues.text);
+            self.view('view');
+        }
+        parent.editing(false);
+        self.messages([]);
+    };
+
+    self.save = function(parent) {
+        self.messages([]);
+        if (self.name() === undefined || self.name().length === 0) {
```

Comment:
Can you use knockout field validation instead to check that the name and citation text is there? https://github.com/Knockout-Contrib/Knockout-Validation#getting-started